draft/0001-organization.md

```diff
@@ -0,0 +1,292 @@
+5. The `exp` claim MUST be greater than or equal to the current Unix time. The resource server MAY add a reasonable leeway to this time.
+6. The `iat` claim MUST be less than or equal to the current Unix time.
+7. The `scope` claim MUST be a subset of the permissions in the organization template.
+
+## 7. Drawbacks
+
+### 7.1. ID token size
+
+The introduction of new claims may increase the size of the ID token. In a multi-tenant environment, the ID token size may significantly increase if the user has numerous memberships and roles.
+
+To mitigate this issue, the OP MAY choose to include organization claims in the ID token only when the client requests the [Userinfo Endpoint](https://openid.net/specs/openid-connect-core-1_0.html#UserInfo) with the `organization` scope.
+
+### 7.2. Allowed dynamic organizations
+
+Unlike Resource Indicators, which require the client to specify the resource identifier when making an authentication request, this specification allows the client to request access tokens for any organization the user is a member of. This is because resource servers typically don't interact directly with the user, but organization memberships are directly linked to the user.
+
+In essence, organization memberships function more like an identity attribute. It is acceptable to request access tokens dynamically for any organization the user is a member of.
+
+From a security standpoint, we have defined an organization template to restrict the organization scopes that the user can request. We also adhere to the OAuth 2.0 specification to ensure that organization scopes must be declared in the authentication request, and the authorization server must not grant any additional scopes for security reasons.
+### 7.3. The absence of resource indicators
+
+[Resource Indicators](https://www.rfc-editor.org/rfc/rfc8707.html) define a standardized way to request access tokens for specific resources, which is particularly useful for multiple API endpoints with varying permissions. However, applying this concept to organizations remains unclear and requires further discussion.
+
+## 8. Rationale and alternatives
+
+The concept of organizations is prevalent in modern SaaS applications, known by various names like workspaces, projects, and more, as seen in Slack, GitHub, and Google Workspace. Many of our users have expressed the need for this feature, making it a crucial addition to Logto.
+
+During discussions, two key aspects emerged:
+
+**Implementation**
+
+Logto Cloud already follows a typical multi-tenant model for identity, and we initially used Resource Indicators to support it. However, we encountered two challenges:
+
+- Duplicating permissions and roles for each tenant (organization) proved to be unscalable and difficult to maintain.
+- There was no direct connection between an identity and the resources the identity has access to, requiring users to perform at least two authentication requests to obtain an access token for a specific tenant.
```

Comment:
nit: should we specify 'two authentication requests'?

Reply:
updated